File: miniapp/src/services/adsApi.ts
import http from '@/api/http';
import type { NearbyAd } from '@/types';

export interface NearbyAdsRequest {
  lat: number;
  lng: number;
  radiusKm?: number;
  categoryId?: string;
  subcategoryId?: string;
  season?: string;
}

export async function fetchNearbyAds(params: NearbyAdsRequest): Promise<NearbyAd[]> {
  const { lat, lng, radiusKm = 5, categoryId, subcategoryId, season } = params;
  const response = await http.get('/api/ads/nearby', {
    params: {
      lat,
      lng,
      radiusKm,
      ...(categoryId ? { categoryId } : {}),
      ...(subcategoryId ? { subcategoryId } : {}),
      ...(season ? { season } : {}),
    },
  });

  const data = response.data;
  if (Array.isArray(data)) {
    return data as NearbyAd[];
  }

  return (data?.items as NearbyAd[]) || [];
}

File: android/app/src/main/java/com/kufarcode/app/data/remote/api/AdsApi.kt
package com.kufarcode.app.data.remote.api

import com.kufarcode.app.data.remote.dto.AdDto
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.Path

interface AdsApi {
    @GET("/api/ads/trending")
    suspend fun getTrending(@Query("cityCode") cityCode: String): List<AdDto>

    @GET("/api/ads/nearby")
    suspend fun getNearby(
        @Query("lat") lat: Double,
        @Query("lng") lng: Double,
        @Query("radiusKm") radiusKm: Double,
        @Query("categoryId") categoryId: String? = null
    ): List<AdDto>

    @GET("/api/ads")
    suspend fun getBySeason(@Query("seasonCode") seasonCode: String): List<AdDto>
}
